src/views/common/Animation/AnimationControls/useUrlPlaybackWindowSupport.ts

```diff
@@ -0,0 +1,69 @@
+import { useUrlState } from 'figurl/UrlStateContext'
+import React, { useCallback } from 'react'
+import { AnimationState, AnimationStateAction } from 'views/common/Animation/AnimationStateReducer'
+
+const useUrlPlaybackWindowSupport = (dispatch: React.Dispatch<AnimationStateAction<any>>) => {
+    /**
+     * Hook returns three callbacks.
+     * 
+     * setStateToInitialUrl sets the window and current frame from data in the initial URL, if populated.
+     * This parameterless callback should be executed by consumer when the animation frame data updates.
+     * 
+     * handleSaveWindowToUrl is a callback that persists elements of the current playback state into
+     * the URL. This callback is likely to be consumed by the PlaybackBookmarkButton.
+     * 
+     * compareStateToUrl takes the current animation state and compares it to the one represented
+     * in the URL. It returns true if they match, else false.
+     * 
+     */
+    const { initialUrlState, updateUrlState, urlState } = useUrlState()
+
+    const setStateToInitialUrl = useCallback(() => {
+        const { playbackWindowFrameStart, playbackWindowFrameEnd, playbackWindowFocusFrame } = initialUrlState
+        if (playbackWindowFrameStart !== undefined && playbackWindowFrameEnd !== undefined) {
+            const minFrame = Math.min(playbackWindowFrameStart, playbackWindowFrameEnd)
+            const maxFrame = Math.max(playbackWindowFrameStart, playbackWindowFrameEnd)
+            dispatch({
+                type: 'SET_WINDOW',
+                bounds: [minFrame, maxFrame]
+            })
+            dispatch({
+                type: 'PROPOSE_WINDOW',
+                bounds: [minFrame, maxFrame]
+            })
+        }
+        if (playbackWindowFocusFrame) {
+            dispatch({
+                type: 'SET_CURRENT_FRAME',
+                newIndex: playbackWindowFocusFrame
+            })
+        }
+    }, [dispatch, initialUrlState])
+
+    const handleSaveWindowToUrl = useCallback((state: AnimationState<any>) => {
```

Comment:
These are callbacks that take the `AnimationState` as a parameter. So we encapsulate the URL state here (centralizing the semantics of what we call those fields) and avoid having other files take dependency on the `useUrlState()` hook. Also, since the URL changes much less often than the animation state, better to have this hook return some long-lived callbacks that don't have to be refreshed when animation state changes.